### PR TITLE
Milestone2 PR

### PR DESCRIPTION
**Pull request by abouhid**
This milestone applies the following changes to the project:

Create a components/actions/containers folder for all the presentational components.
Implements App component
Implements Bookslist component
Implements BookForm component
Implements Book component
Defines the Redux actions and reducers
Initializes the Redux store

### CODE REVIEW

**Comment by abouhid** [comment]
I have no idea why, but the deployment is crashing here on the PR, but in netlify it is working fine: 
https://magicbookspeteralex.netlify.app/
Please, if you do not accept it like this, provide us some guidance on how to fix it...

File: src/pages/index.js [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+    category: "horror",
+  },
+]
+const IndexPage = () => {
+  const [books, setBooks] = useState(bookObj)
+  return (
+    <div>
+      <Provider store={store}>
+        {console.log(books)}
+        <App />
```
Comment (shloch) [inline src/pages/index.js]: remove `console.log` after testing

File: src/components/BookForm.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React from "react"
+import Form from "react-bootstrap/Form"
+import { Button } from "react-bootstrap"
+import { useForm } from "react-hook-form"
```
Comment (shloch) [inline src/components/BookForm.js]: relocate this form to `src/containers` folder (as per the requirements)

File: src/components/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from "react"
+import Book from "./Book"
+import "bootstrap/dist/css/bootstrap.min.css"
+import { bookObj } from "../pages"
+
```
Comment (shloch) [inline src/components/BooksList.js]: relocate this form to `src/containers` folder (as per the requirements)

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @abouhid , @peterrobert 

Essentially everything is great in this project. You nailed the milestone. I however have very minimal corrections to do on your part before a final thumbs up [comments below]

### Happy coding

File: src/containers/BookForm.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import React from "react"
+import Form from "react-bootstrap/Form"
+import { Button } from "react-bootstrap"
+import { useForm } from "react-hook-form"
+
+const BookForm = () => {
+  const { register, handleSubmit, watch, errors } = useForm()
+  const onSubmit = data => {
+    console.log(data)
```
Comment (Torres-ssf) [inline src/containers/BookForm.js]: - [ ] We need to make our code clean before merging this milestone. Therefore we need to remove all console logs. We need to remove this console log here

File: src/reducers/books.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { func } from "prop-types"
+import { createBook, removeBook } from "../actions"
+import { createStore } from "redux"
+import bookReducer from "../reducers"
+
+const store = createStore(bookReducer)
+
+store.subscribe(() => console.log(store.getState()))
```
Comment (Torres-ssf) [inline src/reducers/books.js]: - [ ] We need to remove this console log here

File: src/components/Book.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+// const mapStateToProps = (state) => ({
+
+// })
+
+// const mapDispatchToProps = {
+
+// }
+
+// export default connect(mapStateToProps, mapDispatchToProps)(Book)
```
Comment (Torres-ssf) [inline src/components/Book.js]: - [ ] We need to delete this commented code here.

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi team :hand: 
Great job implementing the requested changes. However there still some changes we need to do.

- [ ] Please let me ask you, was your project bootstrapped with `create-react-app`? I noticed that your project has a lot of additional configuration files. Bootstrap this project with `create-react-app` is a requirement.
Screenshot from 2021-01-20 17-07-27

- [ ] Your `.eslintrc` file is different from the one provided by Microverse. We should not change make changes to this file. We need to use [this one](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json) instead.

Happy coding :computer: 
**Sergio Torres**

**Comment by abouhid** [comment]
I decided to work with react-bootstrap and gatsby to make the most out of this project. I don't think it should be a problem

**Comment by abouhid** [comment]
> ### Changes required 
> Hi team 
> Great job implementing the requested changes. However there still some changes we need to do.
> 
> * [ ]  Please let me ask you, was your project bootstrapped with `create-react-app`? I noticed that your project has a lot of additional configuration files. Bootstrap this project with `create-react-app` is a requirement.
>   Screenshot from 2021-01-20 17-07-27
> * [ ]  Your `.eslintrc` file is different from the one provided by Microverse. We should not change make changes to this file. We need to use [this one](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json) instead.
> 
> Happy coding 
> **Sergio Torres**

I decided to work with react-bootstrap and gatsby to make the most out of this project. I don't think it should be a problem since the sole purpose of all of this is learning.
About the linters, it is necessary to change it for me to use gatsby...

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, congratulations on this milestone :fireworks: Having met all the minimum requirements and having fixed the comments outlined by the previous TSE's :+1: I hereby approve the project :slightly_smiling_face: 

Kindly go ahead and merge into development :+1: Do not forget to submit the project completion form :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**